Base/QTGUI/qSlicerSingletonViewFactory.h

```diff
@@ -0,0 +1,58 @@
+/*==============================================================================
+
+Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+See COPYRIGHT.txt
+or http://www.slicer.org/copyright/copyright.txt for details.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+and was supported through CANARIE's Research Software Program, and Cancer
+Care Ontario.
+
+==============================================================================*/
+
+#ifndef qSlicerSingletonViewFactory_h
+#define qSlicerSingletonViewFactory_h
+
+// CTK includes
+#include "ctkLayoutViewFactory.h"
+
+// QtGUI includes
+#include "qSlicerBaseQTGUIExport.h"
+
+class ctkDICOMBrowser;
+class qSlicerSingletonViewFactoryPrivate;
+
+class Q_SLICER_BASE_QTGUI_EXPORT qSlicerSingletonViewFactory : public ctkLayoutViewFactory
+{
```

Comment:
Add documentation, most importantly, give a high-level summary of why this is useful and how it is supposed to be used.

Reply:
And make sure to define the meaning of "Singleton" in this context.